### ISSUE TITLE
[MNG-1530] Converter for maven 1 projects

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#3462]
**[Fabrizio Giustina](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fgiust)** opened **[MNG-1530](https://issues.apache.org/jira/browse/MNG-1530?redirect=false)** and commented

A simple mvn plugins that converts a maven 1 project.xml to a maven 2 pom.xml.
Apart from the pom v3 to v4 conversion it also adds specific plugin configurations to pom.xml from project.properties (there are a few samples for some compiler, surefire, war settings and it can be easily extended).
After installing it using mvn:install you can run "mvn maven1:convert" on maven 1 project for a quick start on maven 2.

Note: the org.apache.maven.model.v3_0_0.* classes are actually embedded in the plugin due to MNG-1436 and MAVENUPLOAD-583

If there is interest in this I can commit it to svn, please comment/vote here.

---

**Attachments:**
- [maven-maven1-plugin.zip](https://issues.apache.org/jira/secure/attachment/12711447/maven-maven1-plugin.zip) (_39.85 kB_)

**Issue Links:**
- [MRM-29](https://issues.apache.org/jira/browse/MRM-29) Factorize pom conversion out of repoclean

5 votes, 4 watchers

**Comment by jira-importer** [comment]
**[Miguel Griffa](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mgriffa)** commented

IMHO, it is extremly important to get this into maven asap and test and improve.
The ease of migration will bring an important ammount of m1 users to m2, actually, if it is easy to get migrated, they can peridically check if the artifact will work with m2 or not. of course, migrating maven.xml can be tricky... maybe maven-compat-plugin? m2 rocks, I thik that anything that will push it a little more is ok.
just muy 0.02c

**Comment by jira-importer** [comment]
**[Carlos Sanchez Gonzalez](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlos)** commented

The common classes between the plugin and repoclean should be factorized out to a library, see related issue

**Comment by jira-importer** [comment]
**[Grégory Joseph](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gjoseph)** commented

'course this is very interesting. There's also MNG-701 relating this, not sure which one is better, a standalone tool or a plugin. In any case, it IS useful and should be available asap :)
Some more doc for newbies would probly be needed too.

**Comment by jira-importer** [comment]
**[Bryan Pauquette](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bpauquette)** commented

Newbie question.....

How do you install it?

**Comment by jira-importer** [comment]
**[Bryan Pauquette](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bpauquette)** commented

Newbie answer?

mvn install:install-file -DgroupId=group-id -DartifactId=artifact-id -Dversion=version -Dpackaging=packaging -Dfile=fileToInstal

Now I need to know the group-id artifact-d version packaging and fileToInstal.   
1 question leads to 5 new questions.

My ignorance has increased by a factor of 5.

**Comment by jira-importer** [comment]
**[Grégory Joseph](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gjoseph)** commented

Check the pom ;)

**Comment by jira-importer** [comment]
**[Bryan Pauquette](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bpauquette)** commented

Newbie Answer:

1) Download the file to a local directory
2) unzip it.   I can't help you if you don't know how to unzip it..... try google
3) Navigate to the directory where the pom.xml is stored.
4) mvn -version
   Maven version: 2.0.2
   If your computer  doesn't respond with a version number you haven't  installed maven2 correctly yet. Go back and install it.  This time read more carefully........   http://maven.apache.org/download.html#installation
5) type mvn install
6) You have installed it.
7) Navigate to your maven1 project.  Where the project.xml file is.
8) type mvn maven1:convert
9) Watch it die because it didn't parse the id tag correctly because your a newbie.....id is deprecated.
10) Go look up the new tag for id in maven 1.0?

**Comment by jira-importer** [comment]
**[Bryan Pauquette](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bpauquette)** commented

Ok,  so you changed \<id>myproject\</id> to \<artifactId>myproject\</artifactId>.....

Now you need to delete your developer list and contributer lists because they have the \<id>\</id> element also.

Your the last guy coding anyway, everyone else has gone home.   What you need your team for?

Now it works...

**Comment by jira-importer** [comment]
**[corinne krych](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=corinnekrych)** commented

Is it possible to translate a hierarchy of project.xml to pom.xml at once?

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

What's the status on this? I've had a look at the source and it's looking good. Also tried to convert a couple of poms with satisfactory results. Carlos' concern about some refactoring has been solved in MRM-29.

There seems to be some overlap with code in other places in SVN though. I have found these that seems related:
- /maven/components/trunk/maven-model-converter
- /maven/components/trunk/maven-repository-tools
- /maven/sandbox/pom-pre-alpha-converter

From the look of things maven-model-converter is the most recent one. Should this plugin use that component instead of the classes currently in the zip-file?

The code in the zip seems to be working, and there are a lot of questions on the user list about a conversion tool. Would it be possible to find a spot for this plugin in the sandbox so that it can evolve some more? I'd be willing to write a couple of PluginConfigurationConverters.
